src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -488,4 +494,124 @@ public void close() throws Exception {
         });
     }
 
+    // Support methods for REST API's
+
+    public List<String> getInspectionHooksIds() throws Exception {
+
+        return this.txControl.supports(() -> {
+            return this.utils.txInspectionHookEntities().stream().map(InspectionHookEntity::getHookId)
+                    .collect(Collectors.toList());
+        });
+    }
+
+    public Element updateInspectionPort(InspectionPortElement inspectionPort) throws Exception {
+
+        InspectionPortEntity inspectionPortEntity = (InspectionPortEntity) getInspectionPort(inspectionPort);
+        this.utils.throwExceptionIfNullElement(inspectionPortEntity);
+
+        if (inspectionPort.getIngressPort() == null) {
+            this.utils.throwExceptionIfNullElement(inspectionPort.getIngressPort(), "Ingress port is null");
+        }
+        if (inspectionPort.getEgressPort() == null) {
+            this.utils.throwExceptionIfNullElement(inspectionPort.getEgressPort(), "Egress port is null");
+        }
+
+        if (inspectionPort.getEgressPort().getElementId() == inspectionPort.getIngressPort().getElementId()) {
+            NetworkElementEntity element = this.utils
+                    .findNetworkElementEntityByElementId(inspectionPort.getEgressPort().getElementId());
+            this.utils.throwExceptionIfNullElement(element, "Egress port not valid");
+        } else {
+            NetworkElementEntity element = this.utils
+                    .findNetworkElementEntityByElementId(inspectionPort.getEgressPort().getElementId());
+            this.utils.throwExceptionIfNullElement(element, "Egress port not valid");
+            element = this.utils.findNetworkElementEntityByElementId(inspectionPort.getIngressPort().getElementId());
+            this.utils.throwExceptionIfNullElement(element, "Ingress port not valid");
+        }
+
+        return this.txControl.required(() -> {
+            inspectionPortEntity.setEgressPort((NetworkElementEntity) inspectionPort.getEgressPort());
```

Comment:
how is this working? inspectionport is passed in from external, we cannot make an assumption that its a NetworkElementEntity and its in the current session

Reply:
@arvindn05, i am checking for validity of Egress, Ingress port elements which is embedded as a part of inspection port. As I see from definition ingress is a network element type. So checking whether this Id already exists in db. Is this incorrect? Let's follow up offline.

Reply:
how is the `inspectionPort.getEgressPort()` guaranteed to be a entity. From the signature the egrees port is only going to be a NetworkElement so its an usafe cast.

when you set the `inspectionPortEntity.setEgressPort` the port should be guaranteed to be a    NetworkElementEntity AND it has be attached to a session(using merge or some other mechanism)

As this method updateInspectionHook is a public method it can be called from anywhere and i could pass in just any network element and this would cause failure as part of the cast...

Reply:
Thanks I got it